R/surv.b.R

```diff
@@ -40,150 +52,237 @@ survClass <- R6::R6Class(
+                # ------- bind the methods
+                DF <- rbind(dat1,dat2,dat3,dat4)
+                # ------- add the df and the p-values
+                DF$df <- df
+                DF$pvalue <- pchisq(DF$chisqr, df, lower.tail = FALSE)
                 
-                if (length(groups) >= 2) {
-                    groupsData <- list()
-                    for (group in groups) {
-                        ss <- (data$group == group)
-                        groupsData[[group]] <- subset(data, subset=ss, select=c('elapsed', 'event'))
-                    }
-                }
                 
-                for (pair in tt$rowKeys) {
-                    
-                    x <- groupsData[[pair[1]]]
-                    y <- groupsData[[pair[2]]]
+                #################################
+                ### Populates the table with the  
+                ### tests results
+                #################################
+                tt <- self$results$tests
+                if (tt$isNotFilled() && length(self$options$tests) > 0) {
                     
                     for (i in seq_along(self$options$tests)) {
+                        
                         test <- self$options$tests[i]
                         
-                        if (tt$isFilled(rowKey=pair, col=paste0('nu[', test, ']')))
+                        if (tt$isFilled(rowKey=1, col=paste0('chisqr[', test, ']')))
                             next()
                         
                         private$.checkpoint()
-                        
-                        result <- EnvStats::twoSampleLinearRankTestCensored(
-                            test = test,
-                            x = x$elapsed,
-                            x.censored = ! x$event,
-                            y = y$elapsed,
-                            y.censored = ! y$event,
-                            censoring.side = 'right')
-                        
                         row <- list()
+                        row[[paste0('chisqr[', test, ']')]] <- DF$chisqr[i]
+                        row[[paste0('df[', test, ']')]] <- DF$df[i]
+                        row[[paste0('pvalue[', test, ']')]] <- DF$pvalue[i]
                         
-                        row[[paste0('z[', test, ']')]] <- result$statistic['z']
-                        row[[paste0('nu[', test, ']')]] <- result$statistic['nu']
-                        row[[paste0('nuse[', test, ']')]] <- sqrt(result$statistic['var.nu'])
-                        row[[paste0('p[', test, ']')]] <- result$p.value
-                        
-                        tt$setRow(rowKey=pair, values=row)
+                        tt$setRow(rowKey=1, values=row)
                     }
+                    
                 }
+                tt$setNote("note1","Log-Rank and Peto-Peto tests use survdiff from the survival package. Gehan and Tarone-Ware tests are obtained from the coin package.",init=TRUE)
```

Comment:
are you better off using the reference system for this?

Reply:
I think I'm using the reference system now (just used the addFootnote method) and it looks much better.

Reply:
oh, sorry, by reference system, i mean the little reference numbers which appear below tables.

(you possibly turned this feature off from the kebab menu, so you might have forgotten it was there)

so this is how we let people know that the ANOVA uses the car package:

![Screen Shot 2020-06-05 at 11 16 44](https://user-images.githubusercontent.com/3240247/83826012-33e93380-a71e-11ea-98f4-47120eeaf740.png)

you define your references in the `00refs.yaml` file:

https://github.com/jamovi/jmv/blob/master/jamovi/00refs.yaml

and then you attach the to results elements in the .r.yaml like so:

https://github.com/jamovi/jmv/blob/master/jamovi/ancova.r.yaml#L71-L75

(you can even attach them to individual columns. if the column/table isn't visible, then the reference won't be displayed).

and you can set them programmatically too:

https://github.com/jamovi/jmv/blob/master/R/ancova.b.R#L171-L175